lms/djangoapps/edxnotes/helpers.py

```diff
@@ -239,39 +255,104 @@ def get_index(usage_key, children):
     return children.index(usage_key)
 
 
-def search(user, course, query_string):
+def construct_urls(request, course_id, api_next_url, api_previous_url):
     """
-    Returns search results for the `query_string(str)`.
+    Construct next and previous urls for LMS. `api_next_url` and `api_previous_url`
+    are returned from notes api but we need to transform them according to LMS notes
+    views by removing and replacing extra information.
+
+    Arguments:
+        request: HTTP request object
+        course_id: course id
+        api_next_url: notes api next url
+        api_previous_url: notes api previous url
+
+    Returns:
+        next_url: lms notes next url
+        previous_url: lms notes previous url
     """
-    response = send_request(user, course.id, "search", query_string)
-    try:
-        content = json.loads(response.content)
-        collection = content["rows"]
-    except (ValueError, KeyError):
-        log.warning("invalid JSON: %s", response.content)
-        raise EdxNotesParseError(_("Server error. Please try again in a few minutes."))
-
-    content.update({
-        "rows": preprocess_collection(user, course, collection)
-    })
+    def encoded_query_params(url):
+        """
+        Extract and return encoded query params string.
+        """
+        keys = ('page', 'page_size', 'text')
+        parsed = urlparse.urlparse(url)
+        query_params = urlparse.parse_qs(parsed.query)
+
+        return urlencode({key: query_params.get(key)[0] for key in keys if key in query_params})
+
+    base_url = reverse("notes", kwargs={"course_id": course_id})
+    next_url = api_next_url
```

Comment:
We can use `build_absolute_url` too:
```
next_url = "{}?{}".format(request.build_absolute_uri(notes_url), encoded_query_params(api_next_url))
```
https://docs.djangoproject.com/en/1.9/ref/request-response/#django.http.HttpRequest.build_absolute_uri